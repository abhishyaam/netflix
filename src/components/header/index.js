import React from 'react';
import { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  Text,
  FeatureCallOut,
  Link,
  Group,
  Picture,
  Dropdown,
  Profile,
  SearchIcon,
  Search,
  SearchInput,
  PlayButton,
} from './styles/header';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture src={`/images/users/${src}.png`} {...restProps} />;
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSeacrhActive] = useState(false);
  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSeacrhActive((searchActive) => !searchActive)}
      >
        <img src='/images/icons/search.png' alt='Search' />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder='Search films and series'
        active={searchActive}
      />
    </Search>
  );
};
