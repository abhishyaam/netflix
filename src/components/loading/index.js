import React from 'react';
import { Picture, Spinner, LockBody, ReleaseBody } from './styles/loading';

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} data-testid='loading-picture' />
    </Spinner>
  );
}

Loading.ReleaseBody = function LoadingRealeaseBody({ src, ...restProps }) {
  return <ReleaseBody />;
};
