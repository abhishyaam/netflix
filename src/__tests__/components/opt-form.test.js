import React from 'react';
import { render } from '@testing-library/react';

import { OptForm } from '../../components';

describe('<OtpForm />', () => {
  it('renders <OtpForm /> with populated data', () => {
    const { container, getByText, getAllByPlaceholderText } = render(
      <OptForm>
        <OptForm.Input placeholder='Email Address' />
        <OptForm.Button>Try it now</OptForm.Button>
        <OptForm.Break></OptForm.Break>
        <OptForm.Text>Ready to watch?</OptForm.Text>
      </OptForm>
    );
    expect(getByText('Try it now')).toBeTruthy();
    expect(getAllByPlaceholderText('Email Address')).toBeTruthy();
    expect(getByText('Ready to watch?')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
