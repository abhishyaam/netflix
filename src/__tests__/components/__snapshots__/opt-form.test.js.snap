// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OtpForm /> renders <OtpForm /> with populated data 1`] = `
<div
  class="sc-bdvvtL TjGcK"
>
  <input
    class="sc-hKwDye ktAsxR"
    placeholder="Email Address"
  />
  <button
    class="sc-gsDKAQ dejLHC"
  >
    Try it now
    <img
      alt="Try Now"
      src="/images/icons/chevron-right.png"
    />
  </button>
  <div
    class="sc-eCImPb jCGStq"
  />
  <p
    class="sc-dkPtRN gxdhRg"
  >
    Ready to watch?
  </p>
</div>
`;
